server/src/main/java/org/eclipse/openvsx/AdminAPI.java

```diff
@@ -124,48 +137,87 @@ public ResultJson updateSearchIndex(@RequestParam("token") String tokenValue) {
     }
 
     @PostMapping(
-        path = "/admin/namespace-member",
+        path = "/admin/{namespaceName}/change-member",
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResultJson editNamespaceMember(@RequestParam("token") String tokenValue,
-                                         @RequestParam("namespace") String namespaceName,
+    public ResponseEntity<ResultJson> editNamespaceMember(@PathVariable String namespaceName,
                                          @RequestParam("user") String userName,
                                          @RequestParam(required = false) String provider,
                                          @RequestParam String role) {
-        var token = users.useAccessToken(tokenValue);
-        if (token == null) {
-            return ResultJson.error("Invalid access token.");
-        }
-        if (!UserData.ROLE_ADMIN.equals(token.getUser().getRole())) {
-            return ResultJson.error("Administration role is required.");
-        }
         try {
-            return admins.editNamespaceMember(namespaceName, userName, provider, role, token.getUser());
+            var user = admins.checkAdminUser();
+            return new ResponseEntity<>(admins.editNamespaceMember(namespaceName, userName, provider, role, user), HttpStatus.OK);
         } catch (ErrorResultException exc) {
-            return ResultJson.error(exc.getMessage());
+            return getErrorResponse(exc);
         }
     }
 
     @PostMapping(
-        path = "/admin/delete-extension",
+        path = "/admin/{namespaceName}/delete-extension",
         produces = MediaType.APPLICATION_JSON_VALUE
     )
-    public ResultJson deleteExtension(@RequestParam("token") String tokenValue,
-                                      @RequestParam("namespace") String namespaceName,
+    public ResponseEntity<ResultJson> deleteExtension(@PathVariable String namespaceName,
                                       @RequestParam("extension") String extensionName,
                                       @RequestParam(required = false) String version) {
-        var token = users.useAccessToken(tokenValue);
-        if (token == null) {
-            return ResultJson.error("Invalid access token.");
+        try {
+            var user = admins.checkAdminUser();
+            return new ResponseEntity<ResultJson>(admins.deleteExtension(namespaceName, extensionName, version, user), HttpStatus.OK);
+        } catch (ErrorResultException exc) {
+            return getErrorResponse(exc);
         }
-        if (!UserData.ROLE_ADMIN.equals(token.getUser().getRole())) {
-            return ResultJson.error("Administration role is required.");
+    }
+
+    @GetMapping(path = "/admin/{namespaceName}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @CrossOrigin
```

Comment:
This endpoint does not need the `@CrossOrigin` annotation: it's only used by the dashboard, which is either on the same origin or included in the WebConfig CORS configuration.